src/libcprover-cpp/CMakeLists.txt

```
@@ -61,7 +61,13 @@ target_link_libraries(cprover-api-cpp
 # Here we prepare a space-separated list of cmake generators that will resolved in  absolute paths.
 set(DEPENDENCY_TARGETS "")
 foreach(dep ${LIBRARY_DEPENDENCIES})
-    list(APPEND DEPENDENCY_TARGETS "$<TARGET_FILE:${dep}>")
+    if(TARGET ${dep})
+        list(APPEND DEPENDENCY_TARGETS "$<TARGET_FILE:${dep}>")
+    else()
+        find_library(dep_path ${dep})
+        string(REGEX REPLACE "[.]so$" ".a" dep_static_path ${dep_path})
```

Doing this we assume that the dependent library that is found as a shared library (`.so`) also has a static counterpart (`.a`) in the same folder.
When this is not true the build will fail when trying to squash all dependent static libraries together.

I would suggest to check if the static (`.a`) library exists and warn if it doesn't.

---

I'd prefer to keep the failure: if this fails then the resulting Rust API would not be usable.

---

I agree.

What about a failure with a nice message?

```suggestion
        string(REGEX REPLACE "[.]so$" ".a" dep_static_path ${dep_path})
        if (NOT EXISTS "${dep_static_path}")
          message(FATAL_ERROR "libcprover-cpp cannot be linked as dependency ${dep_path} does not have a static correspondent file (${dep_static_path})")
        endif()
```

---

Thank you!